test/e2e/e2e_tests_suite_test.go

```diff
@@ -0,0 +1,222 @@
+		By("generating the commatrix")
+
+		kubeconfig, ok := os.LookupEnv("KUBECONFIG")
+		if !ok {
+			Fail("must set the KUBECONFIG environment variable")
+		}
+
+		config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+
+		oc := configv1client.NewForConfigOrDie(config)
+
+		cs, err := client.New()
+		Expect(err).NotTo(HaveOccurred())
+
+		deployment := types.MNO
+		isSNO, err := isSNOCluster(oc)
+		Expect(err).NotTo(HaveOccurred())
+
+		if isSNO {
+			deployment = types.SNO
+		}
+
+		infra := types.Cloud
+		isBM, err := infrType(oc)
+		Expect(err).NotTo(HaveOccurred())
+
+		if isBM {
+			infra = types.Baremetal
+		}
+
+		epExporter, err := endpointslices.New(cs)
+		Expect(err).ToNot(HaveOccurred())
+
+		commMatrix, err := commatrixcreator.New(epExporter, "", "", infra, deployment)
+		Expect(err).NotTo(HaveOccurred())
+
+		matrix, err := commMatrix.CreateEndpointMatrix()
+		Expect(err).NotTo(HaveOccurred())
+
+		masterMat, workerMat := matrix.SeparateMatrixByRole()
+
+		masterNFT, err := masterMat.ToNFTables()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("apply firewall on master nodes")
+
+		err = applyRulesToNode(cs, "master", masterNFT)
+		Expect(err).ToNot(HaveOccurred())
+
+		if !isSNO {
+			By("apply firewall on worker nodes")
+
+			workerNFT, err := workerMat.ToNFTables()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = applyRulesToNode(cs, "worker", workerNFT)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+})
+
+func applyRulesToNode(cs *client.ClientSet, role string, NFTtable []byte) error {
+	utilsHelpers := utils.New(cs)
+
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	for _, node := range nodesList.Items {
+		nodeRolde, err := types.GetNodeRole(&node)
+		if err != nil {
+			return err
+		}
+
+		if nodeRolde != role {
+			continue
+		}
+
+		ns := "commatrix-firewall-" + role
+		err = utilsHelpers.CreateNamespace(ns)
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if err := utilsHelpers.DeleteNamespace(ns); err != nil {
+				fmt.Printf("failed cleaning namespace %s: %v", ns, err)
+			}
+		}()
+
+		debugPod, err := utilsHelpers.CreatePodOnNode(node.Name, ns, consts.DefaultDebugPodImage)
+		if err != nil {
+			return fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+		}
+
+		defer func() {
+			err := utilsHelpers.DeletePod(debugPod)
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := []string{
+			"bash", "-c",
+			fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
```

Comment:
this will not work on disconnected env and also it's a waste of time for the test multiple by the number of nodes (as we don't do this is parallel here)